src/dfx/src/lib/identity/mod.rs

```diff
@@ -558,7 +559,7 @@ impl Identity {
         match Identity::wallet_canister_id(network, name)? {
             None => {
                 // If the network is not the IC, we ignore the error and create a new wallet for the identity.
-                if !network.is_ic || create {
+                if (!network.is_ic && std::env::var("DISABLE_AUTO_WALLET").is_err()) || create {
```

Comment:
Let's prefix this with `DFX_`:, `DFX_DISABLE_AUTO_WALLET`
```suggestion
                if (!network.is_ic && std::env::var("DFX_DISABLE_AUTO_WALLET").is_err()) || create {
```